packages/react-magma-dom/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,321 @@
+          } else if (breakpointLayout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(true);
+          } else if (breakpointLayout === StepperLayout.summaryView) {
+            setSummaryViewLayout(true);
+          }
+        }
+        if (window.innerWidth > breakpoint || !breakpoint) {
+          if (breakpointLayout === StepperLayout.showLabels) {
+            setShowLabelsLayout(false);
+          } else if (breakpointLayout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(false);
+          } else if (breakpointLayout === StepperLayout.summaryView) {
+            setSummaryViewLayout(false);
+          }
+          if (layout === StepperLayout.showLabels) {
+            setShowLabelsLayout(true);
+          } else if (layout === StepperLayout.hideLabels) {
+            setHideLabelsLayout(true);
+          } else if (layout === StepperLayout.summaryView) {
+            setSummaryViewLayout(true);
+          }
+        }
+        if (showLabelsLayout) {
+          setHideLabelsLayout(false);
+          setSummaryViewLayout(false);
+        } else if (hideLabelsLayout) {
+          setShowLabelsLayout(false);
+          setSummaryViewLayout(false);
+        } else if (summaryViewLayout) {
+          setShowLabelsLayout(false);
+          setHideLabelsLayout(false);
+        }
+      }
+
+      window.addEventListener('resize', handleResize);
+      handleResize();
+
+      return () => window.removeEventListener('resize', handleResize);
+    }, [windowWidth]);
+
+    // Step states
+    const step = React.Children.toArray(children);
+
+    const Steps = React.Children.map(
+      children,
+      (
+        child: React.ReactElement<React.PropsWithChildren<StepProps>>,
+        index
+      ) => {
+        if (child.type === Step) {
+          const stepStatusStyles =
+            currentStep >= index + 1
+              ? StepStatus.complete
+              : currentStep >= index
+              ? StepStatus.active
+              : StepStatus.incomplete;
+
+          const item = React.cloneElement(child, {
+            'aria-current': currentStep === index,
+            children: currentStep === index ? child : null,
+            key: index,
+            isInverse: isInverse,
+            hasHideLabels: hideLabelsLayout,
+            hasSummaryView: summaryViewLayout,
+            stepStatus: stepStatusStyles,
+          });
+
+          return (
+            <>
+              {item}
+              {index !== step.length - 1 && (
+                <StyledSeparator
+                  isInverse={isInverse}
+                  stepStatus={stepStatusStyles}
+                  theme={theme}
+                />
+              )}
+            </>
+          );
+        }
+      }
+    );
+
+    // When summaryView is set to true, this shows one step label and description at a time based on the active step below the Stepper component.
+
+    const StepLabels = React.Children.map(
```

Comment:
This is getting called every time something changes because it's a constant in the body of the file, and that's not necessary because it's actually only used for summary. Also, if there are lots of steps, it's doing a lot of unnecessary looping.

To avoid this, wrap this in an anonymous function. What this will do is only evaluate it when it gets called:
```
    const getSummaryStepLabels = () => {...};
```

Also consider updating the name as it only gets called for summary, so we can be more specific.